service/turn.proxy/src/main/java/org/kaazing/gateway/service/turn/proxy/TurnProxyAcceptHandler.java

```diff
@@ -0,0 +1,170 @@
+        } else {
+            File pwFile = new File(properties.get(PROPERTY_KEY_PASS_FILE));
+            if (!pwFile.isAbsolute()) {
+                throw new TurnProxyException(
+                        String.format("Only absolute files supported for %s, got: %s", PROPERTY_KEY_PASS_FILE, pwFile.getPath())
+                );
+            }
+            sharedSecret = TurnUtils.getSharedSecret(securityContext.getKeyStore(), properties.get(PROPERTY_KEY_ALIAS), pwFile);
+        }
+    }
+
+    @Override
+    protected AbstractProxyHandler createConnectHandler() {
+        return new TurnProxyConnectHandler();
+    }
+
+    @Override
+    public void sessionCreated(IoSession acceptSession) {
+        acceptSession.setAttribute(TURN_STATE_KEY, TurnSessionState.NOT_CONNECTED);
+        acceptSession.getFilterChain().addLast("STUN_CODEC", new StunCodecFilter(currentTransactions, sharedSecret, keyAlgorithm));
+        super.sessionCreated(acceptSession);
+    }
+
+    @Override
+    public void sessionOpened(IoSession acceptSession) {
+        ConnectSessionInitializer sessionInitializer = new ConnectSessionInitializer();
+        ConnectFuture connectFuture = getServiceContext().connect(connectURI, getConnectHandler(), sessionInitializer);
+        connectFuture.addListener(new ConnectListener(acceptSession));
+        super.sessionOpened(acceptSession);
+    }
+
+    @Override
+    public void messageReceived(IoSession session, Object message) {
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug(String.format("Received message [%s] from [%s]", message, session));
+        }
+        if (message instanceof StunProxyMessage) {
+            // Store the incoming username for this transaction, will be reused if generating the message integrity
+            StunProxyMessage stunProxyMessage = (StunProxyMessage) message;
+            stunProxyMessage.getAttributes().stream().filter(attr -> attr instanceof Username).forEach(
+                attr -> {
+                    String transactionId = Base64.getEncoder().encodeToString(stunProxyMessage.getTransactionId());
+                    String username = ((Username) attr).getUsername();
+                    LOGGER.trace(String.format("Storing username in transactionsMap %s -> %s", transactionId, username));
+                    currentTransactions.putIfAbsent(transactionId, username);
+                }
+            );
+            super.messageReceived(session, message);
+        }
+    }
+
+    /*
+     * Initializer for connect session. It adds the processed accept session headers on the connect session
+     */
+    class ConnectSessionInitializer implements IoSessionInitializer<ConnectFuture> {
+
+        @Override
+        public void initializeSession(IoSession session, ConnectFuture future) {
+            session.getFilterChain().addLast("STUN_CODEC", new StunCodecFilter(currentTransactions, sharedSecret, keyAlgorithm));
```

Comment:
if we do stash the currentTransactions to the session then all the constructors will become cleaner